python/src/deltachat/__init__.py

```diff
@@ -11,4 +13,38 @@ def py_dc_callback(ctx, evt, data1, data2):
     CFFI only allows us to set one global event handler, so this one
     looks up the correct event handler for the given context.
     """
-    return _DC_CALLBACK_MAP.get(ctx, lambda *a: 0)
+    callback = _DC_CALLBACK_MAP.get(ctx, lambda *a: 0)
+    # the following code relates to the deltachat/_build.py's helper
+    # function which provides us signature info of an event call
+    event_sig_types = capi.lib.dc_get_event_signature_types(evt)
+    if data1 and event_sig_types & 1:
+        data1 = ffi.string(ffi.cast('char*', data1))
+    if data2 and event_sig_types & 2:
+        data2 = ffi.string(ffi.cast('char*', data2))
+    evt_name = get_dc_event_name(evt)
+    try:
+        ret = callback(ctx, evt_name, data1, data2)
+        if event_sig_types & 4:
+            return ffi.cast('uintptr_t', ret)
+        elif event_sig_types & 8:
+            return ffi.cast('int', ret)
+    except:
+        raise
+        ret = 0
+    return ret
+
+
+def set_context_callback(dc_context, func):
+    _DC_CALLBACK_MAP[dc_context] = func
+
+
+def clear_context_callback(dc_context):
+    _DC_CALLBACK_MAP.pop(dc_context, None)
+
+
+def get_dc_event_name(integer, _DC_EVENTNAME_MAP={}):
```

Comment:
cut but scary.  why not make `_DC_EVENTNAME_MAP` a module-level constant?  It's a bit more conventional